src/buffer_ops/copy.c

```diff
@@ -259,6 +259,12 @@ bool pmix_value_cmp(pmix_value_t *p, pmix_value_t *p1)
         case PMIX_STRING:
             rc = strcmp(p->data.string, p1->data.string);
             break;
+        case PMIX_COMPRESSED_STRING:
+            if (p->data.bo.size != p1->data.bo.size) {
```

Comment:
it isn't - but then that function is never called, and so I didn't bother to waste the time.